### PR TITLE
added focal point functionality

### PR DESCRIPTION
**Pull request by kubiqsk**
When you provide a number values inside $crop array, it will automatically detect them as a focal point and create correctly positioned image

### CODE REVIEW

File: inc/class-core.php [hunk 1]
```diff
@@ -385,9 +428,13 @@ public function get_fly_file_name( $file_name, $width, $height, $crop ) {
 		if ( true === $crop ) {
 			$crop_extension = '-c';
 		} elseif ( is_array( $crop ) ) {
-			$crop_extension = '-' . implode( '', array_map( function( $position ) {
-				return $position[0];
-			}, $crop ) );
+			if( is_numeric( $crop[0] ) ){
+				$crop_extension = '-f' . round( $crop[0] * 100 ) . '_' . round( $crop[1] * 100 );
```
Comment (kubiqsk) [inline inc/class-core.php]: maybe also `floatval()` these `$crop` values?

File: inc/class-core.php [hunk 2]
```diff
@@ -293,7 +293,50 @@ public function get_attachment_image_src( $attachment_id = 0, $size = '', $crop
 			$image_editor = wp_get_image_editor( $image_path );
 			if ( ! is_wp_error( $image_editor ) ) {
 				// Create new image
-				$image_editor->resize( $width, $height, $crop );
+				if( is_numeric( $crop[0] ) ){
```
Comment (kubiqsk) [inline inc/class-core.php]: Here should be also `is_array( $crop )` part of this condition

**Review by sliesensei — CHANGES_REQUESTED** [review]
You should not use `intval( $size[0] );` and `intval( $size[1] );` to get destination width and height. Instead you should use `$width` and `$height` defined above. In case we are using a size name you're trying to get width and height from a string, this causes a division by 0 later.

**Comment by kubiqsk** [comment]
Sorry, I will not cooperate anymore on this, because I can not wait so long and author of the plugin is not responding on forums or here, so I've created my own plugin
https://wordpress.org/plugins/better-image-sizes/

**Pull request closed without merging**